Week 1-Section3

camodesigns opened the pull request:
Pull request for alln the exercises of week 1- section1

camodesigns committed:
Exercise1

camodesigns committed:
Exercise1 Update

camodesigns committed:
Exercise2

camodesigns committed:
Exercise3

camodesigns committed:
Exercise3Updates

camodesigns committed:
Exercise4

camodesigns committed:
Exercise5 and enunciates

camodesigns committed:
Exercise6

camodesigns committed:
Exercise7

camodesigns committed:
UpdateExercise8

UpdateExercise8

camodesigns committed:
Updates Exercise7&8

Changes of structures method creation in the excercise 7 an exercise 8

camodesigns committed:
Exercise9

Methods
String Begin
String Reverse

palicron commented on Section3/Exercise2/Main2.cpp:
```diff
@@ -0,0 +1,103 @@
+	cin >> Answer3;
+	while (Answer3 != 'T' && Answer1 != 'F') {
+		if (FaultCounter3 < 5) {
+			cout << "Invalid Input, try again! " << endl;;
```
Typo two ;

camodesigns committed:
Exercise8 Update

Change method for Profits

camodesigns commented:
![Exercise1](https://user-images.githubusercontent.com/36427649/179099286-963cff85-cb29-43d5-926c-d873aba5f48d.PNG)
![Exercise2](https://user-images.githubusercontent.com/36427649/179099287-0aebbeff-5c7e-4545-a07f-48b808b8e53b.PNG)
![Exercise3](https://user-images.githubusercontent.com/36427649/179099288-dd7f2574-7cbf-45e6-82aa-644d7dd8fa06.PNG)
![Exercise4](https://user-images.githubusercontent.com/36427649/179099290-73bc05ff-1cfd-40bf-851b-a1a5e6bac18d.PNG)
![Exercise5 error Input](https://user-images.githubusercontent.com/36427649/179099291-396f1a25-4791-4d8a-9ae4-f9aaf7cd3d42.PNG)
![Exercise5](https://user-images.githubusercontent.com/36427649/179099292-fc5b2665-d503-4816-95ab-34bc475c7e88.PNG)
![Exercise6](https://user-images.githubusercontent.com/36427649/179099294-a91f4a1d-34a9-4e98-888c-a519f81acee5.PNG)
![Exercise6PressEnterKey](https://user-images.githubusercontent.com/36427649/179099295-a1b5ddc5-baf7-4838-93f3-db46ab233e47.PNG)
![Exercise7](https://user-images.githubusercontent.com/36427649/179099296-5d4bc835-a5cb-4366-a698-b18db509c297.PNG)
![Exercise8](https://user-images.githubusercontent.com/36427649/179099299-9fafcc98-a422-46b6-a92d-08189bd03ec6.PNG)
![Exercise8Errors](https://user-images.githubusercontent.com/36427649/179099300-966378f9-8907-4636-9375-88d9b7b4b777.PNG)
![Exercise9](https://user-images.githubusercontent.com/36427649/179099301-5d18718e-aa8f-4a06-8f36-5f770be8a873.PNG)
![Exercise7Errors](https://user-images.githubusercontent.com/36427649/179099302-de8d2e7b-1569-4371-96fb-9510f7136bcb.PNG)

camodesigns committed:
UpdateExercise8 Section3 Week1

camodesigns committed:
Update exercise2

Eliminate ; and change some variables

JDAlzate commented on Section3/Exercise1/Main.cpp:
```diff
@@ -0,0 +1,28 @@
+	cout << " Please enter a number between 1-10 please" << endl;
+	cin >> Number;
+	int Counter = 0;
+	while (Number > 10 || Number < 1 || (!(cin >> Number))) 
```
Keep in mind that `cin >> Number` will ask for a number again! So you're asking for a number and then if you give valid input, you'll be asked to input something again.

You could do two things here. You can either:
1. Remove the first `cin >> Number` before the while and remove the `cin >> Number` inside the while. That way, you're only asking for input on each run of the while condition.
2. Use cin.good() or cin.fail() instead to check for valid input that was already given.

In both cases though, you should have the check in the condition _before_ anything else. You should check that the input value was correctly added to the character input before checking if the value is 10 or 1.

JDAlzate commented on Section3/Exercise2/Main2.cpp:
```diff
@@ -0,0 +1,103 @@
+int main() {
+	char Answer1, Answer2, Answer3;
+	cout << "Welcome to your quiz!" << endl;
+	cout << "F-False" << endl;//0
```
What are these comments supposed to tell me about the line of code?

JDAlzate commented on Section3/Exercise2/Main2.cpp:
```diff
@@ -0,0 +1,103 @@
+	cout << "Question 1(1pt)" << endl;
+	cout << "1. The meat is fruit?" << endl;
+	cin >> Answer1;
+	while(Answer1!='T'&& Answer1!='F') 
```
Remember to check if the input itself is valid as well. You can use the good() or fail() methods

JDAlzate commented on Section3/Exercise3/Main3.cpp:
```diff
@@ -0,0 +1,29 @@
+	cout<<"Multiples of 10 " << "[";
+	for (int i = 0; i < 101; i++) {
+		if (i % 10 == 0) {
+			int n = 0;
```
Remember that variable names should be descriptive. `i` is acceptable for loop indices because they are universally accepted, but I have no clue what `n` or `o` are supposed to mean. 🤔

JDAlzate commented on Section3/Exercise3/Main3.cpp:
```diff
@@ -0,0 +1,29 @@
+	for (int i = 0; i < 101; i++) {
+		if (i % 10 == 0) {
+			int n = 0;
+			int *Multiples= new int[n];
```
Please refrain from using pointers, references, and dynamic memory until we get to that stage in the program. There's some considerations we need to take into account when using them. Right now your code is performing dangerous logic.

None of these exercises should require you to use pointers.

JDAlzate commented on Section3/Exercise3/Main3.cpp:
```diff
@@ -0,0 +1,29 @@
+				Multiples[o] = i;
+				cout << Multiples[o] <<",";
+			}
+			n += 1;
```
Please review this code. None of what is happening after the `if (i % 10 == 0)` check is necessary. We can go over it in the afternoon.

JDAlzate commented on Section3/Exercise3/Main3.cpp:
```diff
@@ -0,0 +1,29 @@
+			}
+			n += 1;
+		}
+		else if(i%10>0.0) {
```
Be mindful of the instructions :)
They are asking you to NOT use an else statement

JDAlzate commented on Section3/Exercise4/Main4.cpp:
```diff
@@ -0,0 +1,18 @@
+int main() {
+	int sum = 0;
+	cout << "Multiples of 10 " << "[";
+	for (int i = 0; i < 101; i++) {
```
This is not following the instructions. You're looping 100 times, but the loop is not supposed to loop more than 10.

JDAlzate commented on Section3/Exercise5/Main5.cpp:
```diff
@@ -0,0 +1,34 @@
+using namespace std;
+int main()
+{
+	int Number,Addition=0,IntegerCounter=0;
```
Why have Number declared outside the loop if it is only ever used inside the loop? Remember to declare your variables as close to their definition as possible and try to keep them within the same scope.

JDAlzate commented on Section3/Exercise5/Main5.cpp:
```diff
@@ -0,0 +1,34 @@
+	int ArrSize = sizeof(Array) / sizeof(Array[0]);
+	for (int i = 0; i < ArrSize; i++)
+	{
+		cout << "Enter Enter a number if you want:"<<endl ;
```
You wrote Enter twice here.

JDAlzate commented on Section3/Exercise5/Main5.cpp:
```diff
@@ -0,0 +1,34 @@
+		cin >> Number;
+		IntegerCounter += 1;
+		if (!cin.good() ) {
+			cout << "Sorry, invalid Input! " << endl;;
```
There's an extra semicolon here hehe

JDAlzate commented on Section3/Exercise7/Main7.cpp:
```diff
@@ -0,0 +1,92 @@
+#include <ctime>  
+using namespace std;
+bool repeatValues(int arrRepeat[], int size) {
+	for (int pos = 1; pos < size; pos++)
```
Remember to be explicit with your names. I'm guessing that pos is for position, so why not write position? 

However, I can't figure out what posL is supposed to mean. That's why we need to be explicit with our names. Remember that you're not the only one who will ever read your code. Keep the reader in mind!

JDAlzate commented on Section3/Exercise7/Main7.cpp:
```diff
@@ -0,0 +1,92 @@
+	cout << "Welcome to the lottery!" << endl;
+
+
+	int UserValues;
```
Why declare this variable outside of the loop where it's used?

JDAlzate commented on Section3/Exercise7/Main7.cpp:
```diff
@@ -0,0 +1,92 @@
+		int entry = UserSpace + 1;
+		cout << "Please enter your entry number #" << entry << "(1-20)" << endl;
+		cin >> UserValues;
+		for (int pruebaU = 0; pruebaU < ArrUserSize; pruebaU++)
```
Careful with the names, seems like you forgot to change some test

camodesigns replied:
the user only wins if he has the same numbers in the same boxes of the array, which is why I did it with several if's and 
 so I control that I can win more than 10
![Ifs](https://user-images.githubusercontent.com/36427649/179554431-fb9982b7-9126-48ea-afa9-ae91b66d9900.PNG)
f

JDAlzate commented on Section3/Exercise7/Main7.cpp:
```diff
@@ -0,0 +1,92 @@
+	}
+
+	int Profits=0;
+	if (ArrayLotery[0] == ArrayUser[0]) {
```
What if I get two of the numbers correct? I would only get 10 dollars in that case. I can never get more than 10 dollars.

camodesigns replied:
the user only wins if he has the same numbers in the same boxes of the array, which is why I did it with several if's and
so I control that I can win more than 1
![Ifs](https://user-images.githubusercontent.com/36427649/179555104-ef0c550d-ac9d-43ec-80cb-7b5c1c6fbed0.PNG)
0

JDAlzate commented on Section3/Exercise9/Main9.cpp:
```diff
@@ -0,0 +1,65 @@
+using namespace std;
+void printArray(char arr[],int ending)
+{
+    int i;
```
Why have the variable outside of the scope of the loop.

JDAlzate reviewed (changes requested):
Review is up!

camodesigns committed:
Week1-Section3-Exercise1 Updates

while modifictions:
Conditionals
output use
cin eliminated

camodesigns committed:
Week1-Section3-Exercise 2 Updates

Input Verification
Useless comments eliminated

camodesigns committed:
Week1-Section3-Exercise3 Update

Deleted pointers and array, following the observations and the time of the topics

camodesigns committed:
Week1-Section3- Exercise 3 and 4 update

position brackets changed
Exercise 4 Logic changed for following the exercise rules

camodesigns committed:
Week1-Section3-Exercise5 Update

Extra semicolon Deleted, Extra word deleted

camodesigns committed:
Week1-Section3-Exercise7-Updates

Change names Arrais and variables
position of variables used in the for are done

camodesigns committed:
Week1-Section3-Exercise9 Update

Variable is inside of the scope of the loop now

JDAlzate commented on Section3/Exercise1/Main.cpp:
```diff
@@ -9,7 +9,8 @@ int main() {
 	int Counter = 0;
-	while (Number > 10 || Number < 1 || (!(cin >> Number))) 
+
+	while (cin.good()==false || Number > 10 || Number < 1 ) 
```
This is good, but keep in mind that it's more common to check `!cin.good()` rather than `cin.good() == false`, especially since the other way around would be `cin.good() == true` instead of `cin.good()`.

JDAlzate commented on Section3/Exercise1/Main.cpp:
```diff
@@ -18,11 +19,12 @@ int main() {
-		if (
-			Counter >5) {
-			cout<<"No more think loser"<<endl;
+		if (Counter >4)
```
Try to add spaces in between operators. It helps with readability and keeps consistency.

JDAlzate commented on Section3/Exercise2/Main2.cpp:
```diff
@@ -7,17 +7,18 @@ using namespace std;
 	cout << "1. The meat is fruit?" << endl;
 	cin >> Answer1;
-	while(Answer1!='T'&& Answer1!='F') 
+	while(cin.good()|| Answer1!='T'&& Answer1!='F') 
```
Same comment as above applies here.

JDAlzate commented on Section3/Exercise2/Main2.cpp:
```diff
@@ -7,17 +7,18 @@ using namespace std;
+	while(cin.good()|| Answer1!='T'&& Answer1!='F') 
 	{
-		if(FaultCounter <5){
+		if(FaultCounter <5)
```
And here

JDAlzate commented on Section3/Exercise2/Main2.cpp:
```diff
@@ -44,7 +45,8 @@ int main() {
 	cout << "2.The human is composed for 80% water?" << endl;
 	cin >> Answer2;
-	while (Answer2 != 'T' && Answer2 != 'F') {
+	while (cin.good() || Answer2 != 'T' && Answer2 != 'F')
```
You use spaces correctly here. It's much more readable and organized :)

JDAlzate commented on Section3/Exercise3/Main3.cpp:
```diff
@@ -8,18 +8,16 @@ using namespace std;
-			sum += i+1;
+		else if(i%10>0) 
+		{
+			
```
Why have this extra empty line here.

JDAlzate commented on Section3/Exercise7/Main7.cpp:
```diff
@@ -6,11 +6,11 @@ For each number that is matched in the same index, give the user $10.*/
 using namespace std;
 bool repeatValues(int arrRepeat[], int size) {
-	for (int pos = 1; pos < size; pos++)
+	for (int Loteryposition = 1; Loteryposition < size; Loteryposition++)
```
Two comments here:
- Lottery is written with double 't', there's a typo here.
- Prevent coupling your functions from their specific use. This method is supposed to check for repeating numbers in an array, which has nothing to do with a Lottery. With that in mind, your variables inside the function shouldn't be named as if they were related to the Lottery.

JDAlzate replied:
i, j, and k are universally accepted variables for loop iterations and are appropriate for a simple algorithm such as this one. That's an acceptable exception to the naming conventions :)

JDAlzate reviewed (changes requested):
Some small changes are still needed

camodesigns committed:
Week1-Section3-Exercise7 Update

Change profits verification

camodesigns committed:
Week1-Section3-Updates

Organization
Spaces
Terms

JDAlzate commented on Section3/Exercise8/Main8.cpp:
```diff
@@ -48,15 +48,17 @@ int main()
 		for (int pruebaU = 0; pruebaU < ArrUserSize; pruebaU++)
 		{
-			if (ArrayUser[pruebaU] == UserValues) {
+			if (ArrayUser[pruebaU] == UserValues)
```
You're still using `pruebaU` here 👀

JDAlzate reviewed (changes requested):
Just one final small comment!

camodesigns committed:
Week1-Section3-Exercise 1-2-8

Space
Name of Variables

JDAlzate reviewed (approved):
Merge merge merge!
